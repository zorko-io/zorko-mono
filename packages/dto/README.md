# `@zorko/dto`

Date Transfer Object to work with Zorko API

## Usage

```
const { UserDto, TokenDto } = require('@zorko-io/dto');

```
