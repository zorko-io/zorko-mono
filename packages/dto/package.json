{
  "name": "@zorko/dto",
  "version": "1.0.5-alpha.0",
  "description": "Data Transfer Objects for Zorko API",
  "author": "nesterone <igornester@gmail.com>",
  "homepage": "",
  "private": false,
  "license": "ISC",
  "main": "lib/index.js",
  "typings": "lib/index.d.ts",
  "directories": {
    "lib": "lib"
  },
  "files": [
    "lib"
  ],
  "scripts": {
    "build": "tsc",
    "watch": "tsc --watch",
    "test:watch": "jest --config jest.config.json --watch",
    "test": "jest --config jest.config.json",
    "clean": "rimraf lib && rimraf node_modules"
  },
  "publishConfig": {
    "access": "public"
  },
  "dependencies": {
    "@types/node": "^12.6.8",
    "faker": "^4.1.0",
    "livr": "^2.3.0",
    "reflect-metadata": "^0.1.13",
    "rxjs": "^6.5.2",
    "yup": "^0.27.0"
  },
  "devDependencies": {
    "@types/yup": "^0.26.24",
    "jest": "^24.9.0",
    "rimraf": "^2.6.3"
  },
  "gitHead": "6d67667468be8bc801fa02b0a515463a315bf9b5"
}
