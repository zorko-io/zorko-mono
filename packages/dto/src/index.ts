export * from './auth';
export * from './roles';
export * from './users';
export * from './user-profile';
export * from './repository';
export * from './repository-preview';
