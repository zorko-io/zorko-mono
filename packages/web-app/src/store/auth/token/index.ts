export * from './auth.token.presenter';
export * from './auth.token.state';
export * from './auth.token.getter';
