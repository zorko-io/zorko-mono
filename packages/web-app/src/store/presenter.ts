export interface Presenter<S, I> {
    runMutations(): this
    toJS(): S
    toImmutable(): I
}


export interface Getter {}

