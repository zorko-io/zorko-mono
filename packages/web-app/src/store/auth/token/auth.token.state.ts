import { TokenDto } from '@zorko/dto';

export interface AuthTokenState extends TokenDto {
}
