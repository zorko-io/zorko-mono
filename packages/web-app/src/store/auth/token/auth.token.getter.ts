import { Getter } from '../../presenter';
import { Map } from 'immutable';

export class AuthTokenGetter implements Getter {
  readonly userIdPath: string;
  readonly accessKeyPath: string;

  static create() {
    return new AuthTokenGetter();
  }

  constructor() {
    this.accessKeyPath = 'accessKey';
    this.userIdPath = 'userId';
  }

  hasAccessKey = (state: Map<string, any>): boolean => {
    return Boolean(state.get(this.accessKeyPath));
  };

  getUserId = (state: Map<string, any>): string => {
    return state.get(this.userIdPath);
  };
}
