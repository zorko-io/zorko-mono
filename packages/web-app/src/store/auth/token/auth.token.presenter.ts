import { TokenDto } from '@zorko/dto';
import { fromJS, Map } from 'immutable';
import { AbstractImmutablePresenter } from '../../abstract.immutable.presenter';
import { AuthTokenState } from './auth.token.state';
import { AuthTokenGetter } from './auth.token.getter';

export class AuthTokenPresenter extends AbstractImmutablePresenter<AuthTokenState, AuthTokenGetter>{
  static getDefaults() {
    return fromJS({
      accessKey: '',
      userId: ''
    })
  }

  static create(state?: Map<string, any>) {
    const getter = new AuthTokenGetter();

    return new AuthTokenPresenter(
      state || AuthTokenPresenter.getDefaults(),
      getter
    )
  }

  setAccessKey (value: string) {
    return this.addMutation(
      (mutation: Map<string, any>) =>
        mutation.set(this.getter.accessKeyPath, value)
    );
  }

  setUserId (value: string) {
    return this.addMutation(
      (mutation: Map<string, any>) =>
        mutation.set(this.getter.userIdPath, value)
    );
  }

  update(token: TokenDto){
    this.immutable = fromJS(token);
    return this;
  }

  asJS(): AuthTokenState {
     return this.immutable.toJS() as AuthTokenState
   }
}
