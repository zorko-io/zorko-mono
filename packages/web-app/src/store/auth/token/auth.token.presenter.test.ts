import { AuthTokenPresenter } from './auth.token.presenter';
import { TokenDto } from '@zorko/dto';
import { checkImmutableAndJS } from '../../test.helper';

describe('AuthTokenPresenter', () => {
  let presenter: AuthTokenPresenter;
  let nextToken: TokenDto;

  beforeEach(()=> {
    presenter = AuthTokenPresenter.create();
    nextToken = { accessKey: 'zzzzzz', userId:'dfdfdf' };
  });

  it('creates presenter with defaults', () => {
    let expected = presenter.toJS();
    expect(expected).toMatchSnapshot();
    checkImmutableAndJS(presenter)
  });

  it('update access key', () => {
    presenter = presenter.setAccessKey(nextToken.accessKey);

    let expected = presenter.toJS();

    expect(expected).toMatchSnapshot();
    checkImmutableAndJS(presenter)
  });

  it('update user id', () => {
    presenter = presenter.setUserId(nextToken.userId);

    let expected = presenter.toJS();

    expect(expected).toMatchSnapshot();
    checkImmutableAndJS(presenter)
  });

  it('updates token', () => {
    presenter = presenter
      .setAccessKey(nextToken.accessKey)
      .setUserId(nextToken.userId);

    let expected = presenter.toJS();

    expect(expected).toMatchSnapshot();
    checkImmutableAndJS(presenter)
  });
});
