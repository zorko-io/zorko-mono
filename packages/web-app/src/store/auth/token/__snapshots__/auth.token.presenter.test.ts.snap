// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthTokenPresenter creates presenter with defaults 1`] = `
Object {
  "accessKey": "",
  "userId": "",
}
`;

exports[`AuthTokenPresenter update access key 1`] = `
Object {
  "accessKey": "zzzzzz",
  "userId": "",
}
`;

exports[`AuthTokenPresenter update user id 1`] = `
Object {
  "accessKey": "",
  "userId": "dfdfdf",
}
`;

exports[`AuthTokenPresenter updates token 1`] = `
Object {
  "accessKey": "zzzzzz",
  "userId": "dfdfdf",
}
`;
